### DISCUSSION TITLE
Accessing host ip as environment var for production.conf at systemd unite of playframework debian

### DISCUSSION THREAD

**Discussion by mahadi-hossain** [playframework/playframework#11438] [Q&A] [answered]
I have a playframework application what I run as debian file generated by sbt-native-packager. In the production.conf file I need to access `HOST_IP` as environment variable.

But at systemd (when I run sudo systemctl start app.service) app can't access `export HOST_IP=$(hostname -I | awk '{print $1}') ` env variable. I know systemd unit service file can not access environment variable in bashrc or in profile . I know about EnvironmentFile, but that does not support script. I need to create ec2 Amazon Machine Image snap shot that will be used in aws autoscaling. So assigning ip dynamically is necessary. 

Is there any way (in build.sbt or other) that I can pass environment variable to my debian app production.conf which is run by systemctl ?

**Comment by mkurz** [comment, marked as answer]
First, I in production, there is no build.sbt anymore. You are using sbt-native-packer, so the question is which kind of package do you generate? It's very likely Play will get started through a bash script, probably this one here: https://github.com/sbt/sbt-native-packager/blob/d2fd665a5b3dcd2e4f4b15964a98268bfe4b8235/src/main/resources/com/typesafe/sbt/packager/archetypes/scripts/bash-template#L321-L325
As you can see from the lines I referenced you can pass system variables via `-Dkey=val` arguments, or also use set `JAVA_OPTS`.
Maybe also have a look in the sbt-native-packer file , I think there might be options you can set to set up the service file so it passes on the variables to your app.

**Reply by mahadi-hossain** [reply]
Thanks a lot for your kind assistance. Yes I followed that 
at build.sbt. This is added at start-script for production . 
-------------------------------------------------
at build.sbt

```
bashScriptExtraDefines +=
  """
    |host_ip=$(hostname -I | awk '{print $1}')
    |addJava "-DHOST_IP=$host_ip"
    |""".stripMargin
```

**Reply by mkurz** [reply]
So it work? Great :+1:

**Reply by mahadi-hossain** [reply]
It works :)
